### PR TITLE
Attempt to add support for iterate() with SQLite statements

### PR DESCRIPTION
**Pull request by Skywalker13**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

It's an attempt to add support for iterate() with SQLite statements
(like better-sqlite3 https://github.com/WiseLibs/better-sqlite3/blob/master/docs/api.md#iteratebindparameters---iterator)

Related issues:
- Fixes https://github.com/oven-sh/bun/issues/2603
- Fixes https://github.com/oven-sh/bun/issues/8507
- Fixes https://github.com/oven-sh/bun/issues/10925

These changes are based on the `get` and `all` implementations. It's the first time that I see JavaScriptCore stuff. For this feature, I use javascript generators in the wrapper `bun:sqlite`. I suppose that it should be better to implement in JavaScriptCore but honestly I'm not comfortable enough with it.

I hope that is a good start because IMHO it's an essential feature when working with very large databases.

### How did you verify your code works?

- [x] I included a test for the new code, or existing tests cover it
- [x] I ran my tests locally and they pass (`bun-debug test test-file-name.test`)

```
bun-debug test test/js/bun/sqlite/sqlite.test.js
```

### CODE REVIEW

File: src/js/bun/sqlite.ts [hunk 1]
```diff
@@ -154,6 +157,12 @@ class Statement {
     return this.#raw.all();
   }
 
+  *#iterateNoArgs() {
```
Comment (Jarred-Sumner) [inline src/js/bun/sqlite.ts:160]: Maybe worth also implementing `*[Symbol.iterator]()` version of this?

File: test/js/bun/sqlite/sqlite.test.js [hunk 2]
```diff
@@ -587,6 +587,24 @@ it("db.query()", () => {
...
+  }
+  expect(i).toBe(2);
+
+  // iterate (args)
+  i = 0;
+  for (const row of db.query("SELECT * FROM test WHERE name = $name").iterate({ $name: "World" })) {
+    i === 0 && expect(JSON.stringify(row)).toBe(JSON.stringify({ id: 2, name: "World" }));
+    i++;
+  }
+  expect(i).toBe(1);
```
Comment (Jarred-Sumner) [inline test/js/bun/sqlite/sqlite.test.js:615]: Can you add a test that checks what happens if you call iterate, stop partly through, and then call .all()?

**Comment by Jarred-Sumner** [comment]
Thank you for this! Sorry it took a bit of time to merge.

**Comment by Mwni** [comment]
Very nice

**Pull request merged**